tests/Aspire.Hosting.Tests/MSBuildTests.cs

```diff
@@ -124,4 +123,132 @@ the Aspire.AppHost.SDK targets that will automatically add these references to p
         // Ensure a warning is emitted when an AppHost references a Library project
         Assert.Contains("warning ASPIRE004", output.ToString());
     }
+
+    /// <summary>
+    /// Tests that the metadata sources are emitted correctly.
+    /// </summary>
+    [Fact]
+    public async Task ValidateMetadataSources()
+    {
+        var repoRoot = MSBuildUtils.GetRepoRoot();
```

Comment:
Is it possible to refactor these 2 tests so they aren't so duplicated?

Reply:
Sure! I didn't initially because I didn't want to over constrain them, but you're right there is too much duplication. I refactored into a couple of helpers so that common things like building and setting up the targets can be easily shared, while still allowing tests to customize the project(s) they're creating.

If you had something else in mind, please let me know. Thanks!